sqlmesh/core/config/root.py

```diff
@@ -169,10 +170,21 @@ def _normalize_fields_after(cls, values: t.Dict[str, t.Any]) -> t.Dict[str, t.An
         return values
 
     def get_default_test_connection(
-        self, default_catalog: t.Optional[str] = None
+        self,
+        default_catalog: t.Optional[str] = None,
+        default_catalog_dialect: t.Optional[str] = None,
     ) -> ConnectionConfig:
         return self.default_test_connection_ or DuckDBConnectionConfig(
-            catalogs=None if default_catalog is None else {default_catalog: ":memory:"}
+            catalogs=(
+                None
+                if default_catalog is None
+                else {
+                    # transpile catalog name from main connection dialect to DuckDB
+                    exp.parse_identifier(default_catalog, dialect=default_catalog_dialect).sql(
```

Comment:
Nice job thinking through this detail to make sure the default catalog dialect is used to parse it.